ICAFIX/scripts/computeRecleanFeatures.m

```diff
@@ -0,0 +1,273 @@
+        VOLUMESMOOTH=reshape(VOLUMESMOOTH,size(VOLUMESMOOTH,1)*size(VOLUMESMOOTH,2)*size(VOLUMESMOOTH,3),size(VOLUMESMOOTH,4));
+        VolumeSmooth_data=VOLUMESMOOTH(std(VOLUMESMOOTH,[],2)>0,:);
+        VolumeSmooth_data=VolumeSmooth_data./std(Volume_data(:));
+        CIFTIMSMAllSmooth.cdata=CIFTIMSMAllSmooth.cdata./std(CIFTIMSMAll.cdata(:));
+        
+        % structure idx
+        CIFTI_LENGTH=length(CIFTI.cdata);
+        CORTEX_start_idx=min(cifti_diminfo_dense_get_surface_info(CIFTI.diminfo{1}, 'CORTEX_LEFT').ciftilist);
+        CORTEX_LEFT_end_idx=max(cifti_diminfo_dense_get_surface_info(CIFTI.diminfo{1}, 'CORTEX_LEFT').ciftilist);
+        CORTEX_RIGHT_end_idx=min(cifti_diminfo_dense_get_surface_info(CIFTI.diminfo{1}, 'CORTEX_RIGHT').ciftilist);
+
+        CORTEX_end_idx=max(cifti_diminfo_dense_get_surface_info(CIFTI.diminfo{1}, 'CORTEX_RIGHT').ciftilist);
+        CEREBELLUM_start_idx=min(cifti_diminfo_dense_get_volume_structure_info(CIFTI.diminfo{1}, 'CEREBELLUM_LEFT').ciftilist);
+        CEREBELLUM_end_idx=max(cifti_diminfo_dense_get_volume_structure_info(CIFTI.diminfo{1}, 'CEREBELLUM_RIGHT').ciftilist);
+        BRIAN_STEM_start_idx=min(cifti_diminfo_dense_get_volume_structure_info(CIFTI.diminfo{1}, 'BRAIN_STEM').ciftilist);
+        BRIAN_STEM_end_idx=min(cifti_diminfo_dense_get_volume_structure_info(CIFTI.diminfo{1}, 'BRAIN_STEM').ciftilist);
+        
+        row_names=cell(num_comps, 1);
+        probs=zeros(num_comps,1);
+        for k=1:num_comps
+            row_names{k}=[subj '_' fMRIName '_' num2str(k)];
+            probs(k,:)=probability(k);
+            OrigFeatures(k,:)=FeaturesFile(k,:);
+
+            NewFeatures(k,1)=Stats(k,1); %Percent explained variance
+            NewFeatures(k,2)=Stats(k,2); %Percent total variance
+            NewFeatures(k,3)=sum(Powerspectra(PowerspectraScale>0.25,k))./sum(Powerspectra((PowerspectraScale>0.005).*(PowerspectraScale<0.1)==1,k)); %Ratio of power spectrum less than 0.005 to BOLD range
+            NewFeatures(k,4)=sum(Powerspectra(PowerspectraScale<0.005,k))./sum(Powerspectra((PowerspectraScale>0.005).*(PowerspectraScale<0.1)==1,k)); %Ratio of power spectrum greater than 0.25 to BOLD range
+            NewFeatures(k,5)=sum(abs(sICAweighted(sICAnoisy,k)))./sum(abs(sICAweighted(:,k))); %Ratio of amplitude at noisy (top 12.5% variance) timepoints to all timepoints weighted by explained variance
+            NewFeatures(k,6)=sum(abs(sICA(sICAnoisy,k)))./sum(abs(sICA(:,k))); %Ratio of amplitude at noisy (top 12.5% variance) timepoints to all timepoints unweighted
+            NewFeatures(k,7)=sum(abs(CIFTI.cdata(:,k)))./sum(abs(Volume_data(:,k))); %CIFTI/Volume
+            NewFeatures(k,8)=sum(abs(CIFTI.cdata(CORTEX_start_idx:CORTEX_end_idx,k)))./sum(abs(Volume_data(:,k))); %Cerebral Cortex/Volume
```

Comment:
This requires the cortical structures to be adjacent (which was not originally the case in the early days of cifti), and the above logic requires right to be after left.  Neither of these things is required by the cifti standard, so this could break on any cifti files that we don't directly control the ordering of.  The correct way to handle this is to concatenate the lists of cifti indices for the hemispheres into one array, and use that:
```
CORTEX_idx_list = [cifti_diminfo_dense_get_surface_info(CIFTI.diminfo{1}, 'CORTEX_LEFT').ciftilist, cifti_diminfo_dense_get_surface_info(CIFTI.diminfo{1}, 'CORTEX_RIGHT').ciftilist];
...
            NewFeatures(k, 8)=sum(abs(CIFTI.cdata(CORTEX_idx_list, k))) ./ sum(abs(Volume_data(:, k))); %Cerebral Cortex/Volume
```
This has the added benefit of reordering the feature vector to always be left, then right, even if it didn't start that way (though here we are just doing a sum, so it doesn't matter).

We probably control these specific files to be standard grayordinates files, so this fragile min(left), max(right) trick works, but with the amount of copy+paste that goes on in the pipelines, it would be nice to get it right the first time.

As for non-surface data (in this case, subcortical), just use `cifti_diminfo_dense_get_volume_all_info()` to get the list.